Game/MyProject/Game/SceneManagement/RenderPacket.cpp

```diff
@@ -7,70 +7,26 @@
 
 #include "RenderPacket.hpp"
 
-// -----------------------------
-// CONSTRUCTOR(S) AND DESTRUCTOR
-// -----------------------------
-
-// default constructor
-RenderPacket::RenderPacket() :
-    _mesh_instance(nullptr) {
-        _world_view_projection_matrix.Identity();
-    }
-
-// special constructor
-RenderPacket::RenderPacket(std::shared_ptr<MeshInstance> instace_pointer) :
-    _mesh_instance(instace_pointer) {
-        _world_view_projection_matrix.Identity();
-    }
-
-// special constructor
-RenderPacket::RenderPacket(std::shared_ptr<MeshInstance> instace_pointer, const IvMatrix44& matrix) :
-    _mesh_instance(instace_pointer), _world_view_projection_matrix(matrix) {
-
-    }
-
-// destructor
-RenderPacket::~RenderPacket() {
-    
-}
-
-// -----------------------------------
-// PROTECTED FUNCTION(S) AND METHOD(S)
-// -----------------------------------
-
-// sets up the environment for drawing
-void RenderPacket::setUpDrawing() {
+// sets the states and draws the function
+void RenderPacket::draw() {
+    // setting up shader and wvp_matrix
     IvRenderer::mRenderer->SetShaderProgram(_mesh_instance->getShader());
     _mesh_instance->setWorldViewProjectionMatrix(_world_view_projection_matrix);
-}
-
-// --------------------------------
-// PUBLIC FUNCTION(S) AND METHOD(S)
-// --------------------------------
-
-// sets the mesh instance
-void RenderPacket::setMeshInstance(std::shared_ptr<MeshInstance> instace_pointer) {
-    _mesh_instance = instace_pointer;
-}
-
-// sets the world view projection matrix
-void RenderPacket::setWorldViewProjectionMatrix(const IvMatrix44& matrix) {
-    _world_view_projection_matrix = matrix;
-}
-
-// returns the mesh instance
-std::shared_ptr<MeshInstance> RenderPacket::getMeshInstance() {
-    return _mesh_instance;
-}
-
-// returns the world view projection matrix
-IvMatrix44 RenderPacket::getWorldViewProjectionMatrix() {
-    return _world_view_projection_matrix;
-}
-
-// draws the geometry
-void RenderPacket::Draw() {
-    setUpDrawing();
     
-    IvRenderer::mRenderer->Draw(kTriangleStripPrim, _mesh_instance->getMesh()->getVertexBuffer(), _mesh_instance->getMesh()->getIndexBuffer());
+    // setting up the states
+    if (_use_depth)
+        IvRendererOGL::mRenderer->SetDepthWrite(true);
+    else
+        IvRendererOGL::mRenderer->SetDepthWrite(false);
+    
+    if (_use_blend)
+        IvRendererOGL::mRenderer->SetBlendFunc(kSrcAlphaBlendFunc, kOneMinusSrcAlphaBlendFunc, kAddBlendOp);
+    else
+        IvRendererOGL::mRenderer->SetBlendFunc(kOneBlendFunc, kZeroBlendFunc, kAddBlendOp);
```

Comment:
It seems that this turns blending on/off indeed. It would be nice to note on a comment that turning blending on/off is not needed since it's handled inside the renderer. Usually rendering apis require specific turning on/off of the blending operation.